platform/android/audio_driver_opensl.cpp

```diff
@@ -220,7 +222,11 @@ Error AudioDriverOpenSL::init_input_device() {
 		SL_PCMSAMPLEFORMAT_FIXED_16,
 		SL_PCMSAMPLEFORMAT_FIXED_16,
 		SL_SPEAKER_FRONT_CENTER,
-		SL_BYTEORDER_LITTLEENDIAN
+#ifdef BIG_ENDIAN_ENABLED
+		SL_BYTEORDER_BIGENDIAN,
+#else
+		SL_BYTEORDER_LITTLEENDIAN,
+#endif
```

Comment:
Matches the implementation for creating an output device:
https://github.com/godotengine/godot/blob/32eba1ee17555fbc4f4d348a568b6d9d391de9a5/platform/android/audio_driver_opensl.cpp#L137-L141